File: binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        nums.sort()
        minVal = 0
        maxVal = len(nums)-1
        while(minVal<=maxVal):
            mid = (minVal + maxVal)//2
            if(nums[mid] == target):
                return mid
            if(nums[mid]>target):
                maxVal = mid-1
            else:
                minVal = mid+1
        return -1

File: binary_search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int mi = 0;
        int ma = nums.size()-1;
        int mid = (mi+ma)/2;
        sort(nums.begin(), nums.end());
        while(mi<=ma){
            mid = (mi+ma)/2;
            if(nums[mid]==target){
                return mid;
            }
            if(nums[mid]>target){
                ma = mid-1;
            }
            else{
                mi = mid+1;
            }
        }
        return -1;

    }
};